### DISCUSSION TITLE
Add ui.icon_picker

### DISCUSSION THREAD

**Discussion by khannover** [zauberzeug/nicegui#3907] [Ideas / Feature Requests]
Hi,
i would love to see a ui.icon_picker to select an icon from a list of available (material-)icons.

Currently i use a ui.select element with all the material icon names as options (around 2500).
That's a very long list and i can only show the icon names in the list but not how the icon looks like.

It could work and look like the [ui.color_picker](https://nicegui.io/documentation/color_picker#customize_the_color_picker) but with icons instead of colors of course ;-)

![image](https://github.com/user-attachments/assets/3d4c718c-a6a8-4367-9baf-4f32dd931fce)

I would use it somehow like this:

```python
icon_picker = ui.icon_picker(on_pick=lambda e: ui.notify(f'you picked {e.icon}')).bind_value(config, 'icon_name')
```

**Comment by falkoschindler** [comment]
That's an interesting idea, @khannover.
What would be an application for such an element?

**Reply by khannover** [reply]
I am building a web app, where users can build and customize dashboards. The user has the option to define a dashboard icon, which will then be shown in a sidebar as link, along other dashboard icons.

**Comment by falkoschindler** [comment]
By the way, you can customize the "option" slot of `ui.select` to show icons next to labels:
```py
ui.select([
    'home',
    'settings',
    'info',
], on_change=lambda e: ui.notify(f'You chose "{e.value}".')).add_slot('option', '''
    <q-item :props="props" clickable @click="$parent.$parent.$parent.$parent.toggleOption(props.opt)">
        <q-item-section avatar>
            <q-icon :name="props.label" />
        </q-item-section>
        <q-item-section>
            <q-item-label v-html="props.label"></q-item-label>
        </q-item-section>
    </q-item>
''')
```

But please don't ask about the chain of `$parent`... Somehow we need to traverse from the QItem up to the QSelect component. Maybe there's a better way I'm not aware of.

**Reply by khannover** [reply]
Thank you, thats great. I think im gonna use that as a workaround. But i think a built-in icon_picker which already knows all the icon names would still be a good idea. 

BTW, i have seen that material icons became deprecated and you should use material symbols instead. Are there any plans for supporting those in NiceGUI?

**Reply by falkoschindler** [reply]
Yes, we are aware of Material Icons being deprecated. There is an old PR #2613 which should simplify the use of icons in NiceGUI, but hasn't been finished yet. It also considered dropping support for Icons in favor of Symbols. But as far as we understand, in the frontend Symbols still need to be specified with a "sym_" prefix. NiceGUI could add the prefix automatically, so that `ui.button(icon='home')` renders the icon as, e.g., "sym_o_home" Symbol. But if anyone specifies icons via props like `.props('icon=home')` or `.props('unchecked-icon=visibility_off')`, it's much harder to detect and replace the icon reliably. Therefore we decided not to change anything about Icons and Symbols yet.

**Reply by falkoschindler** [reply]
> Are there any plans for supporting those in NiceGUI?

Even though you still need the right prefix, they are already supported:
https://nicegui.io/documentation/icon#material_icons_and_symbols
